minique/utils/redis_list.py

```diff
@@ -21,18 +21,22 @@ def read_list(
     :param last_n: Attempt to only read the last N lines.
     :return:
     """
-    if not redis_conn.exists(key):
+    # LLEN returns zero for a non-existent key,
+    # and there would be nothing to iterate from an empty
+    # list anyway, so we can do one query less.
+    list_len = redis_conn.llen(key)
+    if list_len == 0:
         return
```

Comment:
was pondering that being less atomic (caching the length and not fetching it each time it's needed) might cause some concurrency issues with multiple readers buuuut I doubt it as they are quite close to each other in any case

Reply:
Yeah, I think it's okay - we don't guarantee that this will read the list to end if it grows while reading.